src/Request/Request.php

```diff
@@ -0,0 +1,258 @@
+     */
+    private function send($uri, $method, array $arg_params = [])
+    {
+        $host = $this->getConfig()->get('host');
+        $extra_params = [
+            'headers' => [
+                'User-Agent' => $this->userAgent(),
+                'Content-type' => 'application/json',
+            ],
+            RequestOptions::VERIFY => (strpos($host, 'onebox') === false),
+        ];
+
+        if ((!isset($arg_params['absolute_url']) || !$arg_params['absolute_url'])
+            && $session = $this->session()->get('session', false)
+        ) {
+            $extra_params['headers']['Authorization'] = "Bearer $session";
+        }
+        $params = array_merge_recursive($extra_params, $arg_params);
+        if (isset($params['form_params'])) {
+            $params['json'] = $params['form_params'];
+            unset($params['form_params']);
+        }
+        $params[RequestOptions::VERIFY] = (strpos($host, 'onebox') === false);
+
+        $client = new Client(
+            [
+                'base_uri' => $this->getBaseUri(),
+                'cookies' => $this->fillCookieJar($params),
+            ]
+        );
+        unset($params['cookies']);
+
+        $this->logger->debug(
+            "#### REQUEST ####\nParams: {params}\nURI: {uri}\nMethod: {method}",
+            [
+                'params' => json_encode($params),
+                'uri' => $uri,
+                'method' => $method,
+            ]
+        );
+
+        //Required objects and arrays stir benign warnings.
+        error_reporting(E_ALL ^ E_WARNING);
+        $request = new HttpRequest(ucwords($method), $uri, $params);
+        error_reporting(E_ALL);
+        $response = $client->send($request, $params);
+
+        return $response;
+    }
+
+    /**
+     * Sets up and fills a cookie jar
+     *
+     * @param array $params Request data to fill jar with
+     * @return \GuzzleHttp\Cookie\CookieJar $jar
+     */
+    private function fillCookieJar(array $params)
+    {
+        $jar = new CookieJar();
+        $cookies = [];
+        if (isset($params['cookies'])) {
+            $cookies = array_merge($cookies, $params['cookies']);
+        }
+        $jar->fromArray($cookies, '');
+        return $jar;
+    }
+
+    /**
+     * Parses the base URI for requests
+     *
+     * @return string
+     */
+    private function getBaseUri()
+    {
+        $base_uri = sprintf(
+            '%s://%s:%s',
+            $this->getConfig()->get('protocol'),
+            $this->getConfig()->get('host'),
+            $this->getConfig()->get('port')
+        );
+        return $base_uri;
+    }
+
+    /**
+     * Gives the user-agent string
+     *
+     * @return string
+     */
+    private function userAgent()
+    {
+        $agent = sprintf(
+            'Terminus/%s (php_version=%s&script=%s)',
+            $this->getConfig()->get('version'),
+            phpversion(),
+            $this->getConfig()->get('script')
```

Comment:
This is the sort of thing where I would expect that just accessing the `config` object directly to obtain various diverse settings is preferable to making a ton of getters / setters in the class.  The getter / setter route is better if there are only a few config items needed.  For the Session class, maybe keeping a reference to the config object is better.